### PR TITLE
Derek Cruz | node notes cli.

### PR DESCRIPTION
**Pull request by Derek-Cruz**
Node-notes-cli

### CODE REVIEW

File: node-notes-cli/writefile.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const data = require('./data.json');
```
Comment (scbowler) [inline node-notes-cli/writefile.js:1]: Here you shouldn't be requiring in the data from the json file.

File: node-notes-cli/writefile.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+const fs = require('fs');
+
+function writeFileReuse() {
```
Comment (scbowler) [inline node-notes-cli/writefile.js:4]: `writeFileReuse` should have  a single parameter of `data` which will be the object you want to save to the file.

File: node-notes-cli/add.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+const data = require('./data.json');
+const writeFileReuse = require('./writefile');
+const obj = data.notes;
+
+function create(string) {
+  obj[data.nextId] = string;
+  data.nextId = data.nextId + 1;
+  writeFileReuse(create);
```
Comment (scbowler) [inline node-notes-cli/add.js:8]: When you call `writeFileReuse` you should be passing it the `data` object, that is the data you want to write to the file. Not the function name. This applies everywhere you are calling `writeFileReuse`

File: node-notes-cli/data.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "nextId": 3,
+  "notes": {
+    "1": "Hi Scott!"
```
Comment (scbowler) [inline node-notes-cli/data.json:4]: Hello 👋

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good, but some updates needed. Let me know if you have any questions.

**Comment by Derek-Cruz** [comment]
Node-notes-cli-update

**Pull request merged**